lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java

```diff
@@ -554,24 +556,32 @@ void fetchVerifiableCredential_whenCalledAgainstExperianKbvCri_retrievesAValidVc
                         getCriOAuthSessionItem());
 
         // Assert
-        var verifiableCredentialJwtValidator = getVerifiableCredentialJwtValidator();
+        var verifiableCredentialJwtValidator = getVerifiableCredentialValidator();
         verifiableCredentialResponse
                 .getVerifiableCredentials()
                 .forEach(
                         credential -> {
                             try {
-                                verifiableCredentialJwtValidator.validate(
-                                        credential, credentialIssuerConfig, TEST_USER);
+                                var vc =
+                                        verifiableCredentialJwtValidator.validate(
+                                                TEST_USER,
+                                                null,
```

Comment:
Ok, I've traced this through and I understand why the value doesn't matter. However I do find it odd that you pass a value to a method called `validate` that isn't used in the validation. I think I'd feel happier if this method was `parse` or similar instead, then it's clearer that it is converting the credential into an object - and as part of that it is also validating it.

(I can't comment on the relevant bit of code as it hasn't changed in this PR so commenting here instead)

Reply:
I could call it `parse` but I might prefer `validateAndParse`. You're right in that it would explain the operation more clearly. I would like to reference the fact it's validating as part of the process though - what do you think?

Reply:
`and` in function names is often a bit of a red flag so I'd be careful there. Parsing requires some validation (otherwise the parsing will fail), but you're right that it doesn't stress that aspect, which is actually quite big.

Reply:
Hmm, it's tricky because some bits like the signing key aren't strictly needed to parse the JWT. That would imply that you should validate the JWT first, but then you'd need to do more validation later on anyway.
Perhaps we should split the validation in two - first check the signature, then validate the JWT contents separately. I'm torn on whether signature checking should be bundled together with the rest of this. On one hand it means we won't ever forget to do it, on the other hand it feels like a separate step.

Reply:
I understand your point on the naming, though I think it wouldn't be too consequential whatever naming we choose though I think splitting the method would probably extend this PR's lifespan too much because of its affect on unit tests... what do you think?

Reply:
I'm fine not to do this in this PR, but I think it's something worth considering more in future work